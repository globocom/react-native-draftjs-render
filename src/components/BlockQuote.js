/*
 * Copyright (c) 2017, Globo.com (https://github.com/globocom)
 *
 * License: MIT
 */

// @flow

import React from 'react';
import {
  StyleSheet,
  View,
} from 'react-native';

import DraftJsText from '../components/DraftJsText';

import type { BlockQuotePropsType } from './types';

const styles = StyleSheet.create({
  blockquoteContainer: {
    borderLeftColor: '#eee',
    borderLeftWidth: 4,
    borderStyle: 'solid',
    marginTop: 22,
    marginBottom: 22,
    paddingLeft: 12,
  },
});

const BlockQuote = (props: BlockQuotePropsType): any => {
  const blockquoteCustomStyleContainer = props.customStyles ?
    props.customStyles.blockquoteContainer :
    undefined;
  const blockquoteCustomStyleIconBefore = props.customStyles ?
    props.customStyles.blockquoteIconBefore :
    undefined;
  const blockquoteCustomStyleIconAfter = props.customStyles ?
    props.customStyles.blockquoteIconAfter :
    undefined;

  return (
    <View style={[styles.blockquoteContainer, blockquoteCustomStyleContainer]}>
      <View style={blockquoteCustomStyleIconBefore} />
      <DraftJsText
        {...props}
      />
      <View style={blockquoteCustomStyleIconAfter} />
    </View>);
};

BlockQuote.defaultProps = {
  customStyles: undefined,
  type: '',
};

export default BlockQuote;
