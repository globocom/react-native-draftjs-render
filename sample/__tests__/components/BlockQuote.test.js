/*
 * Copyright (c) 2017, Globo.com (https://github.com/globocom)
 *
 * License: MIT
 */

/* eslint-env jest */

import 'react-native';

import React from 'react';
import renderer from 'react-test-renderer';

import BlockQuote from '../../../src/components/BlockQuote';

it('renders correctly with a blockquote', () => {
  const text = 'Hello World';
  const tree = renderer.create(<BlockQuote
    type="blockquote"
    text={text}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders null without a blockquote', () => {
  const tree = renderer.create(<BlockQuote />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('extends a style with a customStyle', () => {
  const text = 'Hello World';
  const customStyles = {
    blockquote: {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
    },
  };
  const tree = renderer.create(<BlockQuote
    type="blockquote"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
