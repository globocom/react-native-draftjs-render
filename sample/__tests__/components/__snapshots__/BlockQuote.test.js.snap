// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extends a style with a customStyle 1`] = `
<View
  style={
    Array [
      Object {
        "borderLeftColor": "#eee",
        "borderLeftWidth": 4,
        "borderStyle": "solid",
        "marginBottom": 22,
        "marginTop": 22,
        "paddingLeft": 12,
      },
      undefined,
    ]
  }
>
  <View
    style={undefined}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontStyle": "italic",
          "fontWeight": "normal",
          "marginLeft": 16,
        },
        Object {
          "fontSize": 18,
          "fontWeight": "normal",
          "letterSpacing": -0.75,
          "lineHeight": 32,
        },
      ]
    }
  >
    Hello World
  </Text>
  <View
    style={undefined}
  />
</View>
`;

exports[`renders correctly with a blockquote 1`] = `
<View
  style={
    Array [
      Object {
        "borderLeftColor": "#eee",
        "borderLeftWidth": 4,
        "borderStyle": "solid",
        "marginBottom": 22,
        "marginTop": 22,
        "paddingLeft": 12,
      },
      undefined,
    ]
  }
>
  <View
    style={undefined}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontStyle": "italic",
          "fontWeight": "normal",
          "marginLeft": 16,
        },
        undefined,
      ]
    }
  >
    Hello World
  </Text>
  <View
    style={undefined}
  />
</View>
`;

exports[`renders null without a blockquote 1`] = `
<View
  style={
    Array [
      Object {
        "borderLeftColor": "#eee",
        "borderLeftWidth": 4,
        "borderStyle": "solid",
        "marginBottom": 22,
        "marginTop": 22,
        "paddingLeft": 12,
      },
      undefined,
    ]
  }
>
  <View
    style={undefined}
  />
  <View
    style={undefined}
  />
</View>
`;
